index.js

```diff
@@ -0,0 +1,77 @@
+'use babel';
+
+import path from 'path';
+import { execNode, find } from 'atom-linter';
+
+const GRAMMAR_SCOPES = [
+  'text.html.angular',
+  'text.html.basic',
+  'text.html.erb',
+  'text.html.gohtml',
+  'text.html.jsp',
+  'text.html.mustache',
+  'text.html.handlebars',
+  'text.html.ruby'
+];
+
+export const config = {
+  executablePath: {
+    title: 'Executable Path',
+    description: 'HTMLHint Executable Path',
+    type: 'string',
+    default: path.join(__dirname, '..', 'node_modules', 'htmlhint', 'bin', 'htmlhint')
+  }
+};
+
+const executablePath = () => atom.config.get('linter-htmlhint.executablePath');
+
+export const activate = () => {
+  require('atom-package-deps').install('linter-htmlhint');
+};
+
+export const provideLinter = () => {
+  return {
+    name: 'htmlhint',
+    grammarScopes: GRAMMAR_SCOPES,
+    scope: 'file',
+    lintOnFly: true,
+    lint: editor => {
+      const text = editor.getText();
+      const filePath = editor.getPath();
+
+      if (!text) {
+        return [];
+      }
+
+      const parameters = [filePath, '--format', 'json'];
+      const htmlhintrc = find(filePath, '.htmlhintrc');
```

Comment:
You should figure out the directory of `filePath` and pass that in as the first argument, not the file directly. (Although this might work, it's an extra level of recursion to run through before returning.)